src/AutoMapper.Extensions.ExpressionMapping/MapperExtensions.cs

```diff
@@ -265,12 +265,29 @@ private static bool HasUnderlyingType(this Type type)
 
         private static void AddUnderlyingTypes(this Dictionary<Type, Type> typeMappings, IConfigurationProvider configurationProvider, Type sourceType, Type destType)
         {
-            typeMappings.DoAddTypeMappings
-            (
-                configurationProvider,
-                !sourceType.HasUnderlyingType() ? new List<Type>() : ElementTypeHelper.GetElementTypes(sourceType).ToList(),
-                !destType.HasUnderlyingType() ? new List<Type>() : ElementTypeHelper.GetElementTypes(destType).ToList()
-            );
+            if (sourceType.IsNullableType() || destType.IsNullableType())
```

Comment:
Not a fan of  adding literal types to `typeMappings`.  Is there a chance of unwanted conversions?

Reply:
Since we're expressly checking below for a user-defined mapping between the two... It seems unlikely

Reply:
Probably need more tests.  I suspect if you're mapping a LINQ select for a class like [this one](https://github.com/mjameson-se/AutoMapper.Extensions.ExpressionMapping/blob/d0a992c45973013739928226b3111ceaa95ecbb5/tests/AutoMapper.Extensions.ExpressionMapping.UnitTests/MappingMemberInitWithCustomExpressions.cs#L110)  say you want some integers converted but not others then there may be problems.